daemon/restapi/api_limits_test.go

```diff
@@ -0,0 +1,59 @@
+package restapi
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/rate"
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+	timeRate "golang.org/x/time/rate"
+)
+
+// TestRateLimiterConfigFlag checks that rateLimiterConfig is properly parsed from
+// command-line flag.
+func testRateLimiterConfig(t *testing.T, setConfig func(h *hive.Hive, limiter string, limits string)) {
+	var limiterSet *rate.APILimiterSet
+	take := cell.Invoke(func(l *rate.APILimiterSet) { limiterSet = l })
+
+	h := hive.New(rateLimiterCell, take)
+
+	setConfig(h, APIRequestEndpointCreate, "rate-limit:42/m,rate-burst:1234")
+	assert.Nil(t, h.Start(context.TODO()))
+
+	l := limiterSet.Limiter(APIRequestEndpointCreate)
+	assert.NotNil(t, l)
+
+	p := l.Parameters()
+	assert.Equal(t, timeRate.Limit(42.0/60.0), p.RateLimit)
+	assert.Equal(t, 1234, p.RateBurst)
```

Comment:
:tada: I was going to ask about the StringToString change and whether that broke CI, but this seems like it's now testing the user configuration format and validating it doesn't break.